library/vulnerabilities/ssrf/checkContextForSSRF.ts

```diff
@@ -40,12 +40,19 @@ export function checkContextForSSRF({
     // We have to check the port as well, because the hostname can be the same but with a different port
     // If Node.js is exposed to the internet, we can't be sure about the Host header
     const baseURL = tryParseURL(context.url);
-    if (
-      baseURL &&
-      baseURL.hostname === hostname &&
-      getPortFromURL(baseURL) === port
-    ) {
-      return;
+    if (baseURL && baseURL.hostname === hostname) {
+      const baseURLPort = getPortFromURL(baseURL);
+      if (baseURLPort === port) {
+        return undefined;
+      }
+      // Special case for HTTP/HTTPS ports
+      // In production, the app will be served on port 80 and 443
+      if (baseURLPort === 80 && port === 443) {
```

Comment:
would put this code all in one place right, if we were to change, now its 2x same code

Reply:
Good idea, thanks!